keras/utils/io_utils.py

```diff
@@ -166,6 +168,13 @@ def ask_to_proceed_with_overwrite(filepath):
     return True
 
 
+def is_path_instance(path):
+    # We can't use isinstance here because it would require
```

Comment:
Why not attempt to import `pathlib2` and catch the `ImportError`? If it fails to import, the object can't possibly originate from that library.

Reply:
Yes! thought about this as well... but since the diff has already grown a bit I'm leaving this for now (now it is clear that I've just moved out the function unchanged). But I'll move it back as a private method H5Dict.

Reply:
or just made it "private to the module" for now.